Automattic/simplenote-ios
# Notes List: Tags UI

### Details:
This PR is all about Tags UI:

- New UITableViewCell subclass for Tag Results!
- And we're also getting a new SectionHeader, to be displayed when we're in Search Mode

cc @aerych (Thanks in advance Eric!!!)
Ref. #507

### Test: Regular Results!
1. Log into your account
2. Go thru your tags!

- [x] Verify no Section Header is displayed
- [x] Verify all the things look as expected!

### Test: Search Mode!
1. Log into your account
2. Press over the Search Bar
3. Enter a keyword that yields Tags + Notes

- [x] Verify that Tags look correctly
- [x] Verify that Tag rows are smaller than Note rows!
- [x] Verify that the Tag Section has a cool header that reads `Search by Tag:`
- [x] Verify that the Note Section has a header that reads `Notes`

### Test: Zero Tag Results
1. Press over the Search Bar
2. Enter a result that yields zero tags (but some notes!)

- [x] Verify the Tags Section is completely hidden
- [x] Verify that pressing over any of the Notes pushes the Editor

### Test: Empty State
1. Press over the Search Bar
2. Enter a result that yields zero results (no Tags nor Notes)

- [x] Verify that the legend **No results** shows up onScreen

### Release
`RELEASE-NOTES.txt` was updated in 0ecea89 with:

> Search results now display matching tags!

---

### Screenshot Time!
<img src="https://user-images.githubusercontent.com/1195260/73093708-d74b5180-3ebd-11ea-804e-86e917c86045.png" width="300">
